be1-go/channel/federation/federation.go

```diff
@@ -369,9 +376,57 @@ func (c *Channel) processFederationChallenge(msg message.Message,
 
 	remoteOrg.state = WaitResult
 	remoteOrg.socket = s
-	// Send Federation result to S1
-	// c.remoteServer.Send(...)
 
+	federationResultData := messagedata.FederationResult{
+		Object:       messagedata.FederationObject,
+		Action:       messagedata.FederationActionResult,
+		Status:       "success",
+		PublicKey:    remoteOrg.signedOrganizerPk,
+		ChallengeMsg: remoteOrg.challengeMsg,
+	}
+
+	dataBytes, err := json.Marshal(federationResultData)
+	if err != nil {
+		return xerrors.Errorf("failed to marshal federation result message data: %v", err)
+	}
+
+	dataBase64 := base64.URLEncoding.EncodeToString(dataBytes)
+	signatureBytes, err := c.hub.Sign(dataBytes)
+	if err != nil {
+		return xerrors.Errorf("failed to sign federation result message: %v", err)
+	}
+	signatureBase64 := base64.URLEncoding.EncodeToString(signatureBytes)
+
+	federationResultMsg := message.Message{
+		Data:              dataBase64,
+		Sender:            c.hub.GetPubKeyServ().String(),
```

Comment:
c.hub.GetPubKeyServ().String() returns an hexadecimal representation of the key.
It should be the base64 representation of the key